be/src/common/exception_stack_trace.h

```diff
@@ -0,0 +1,238 @@
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstringop-truncation"
+                strncpy(vProcedureName, vDemangledProcedureName, LIBUNWIND_MAX_PROCNAME_LENGTH);
+#pragma GCC diagnostic pop
+                // Free the memory from __cxa_demangle():
+                free(vDemangledProcedureName);
+            } else {
+                // NOTE: if the demangle fails, we do nothing, so the
+                // non-demangled name will be printed. Thats ok.
+            }
+        } else if (vUnwindStatus == UNW_ENOMEM) {
+            // NOTE: libunwind could resolve the name, but could not store
+            // it in a buffer of only LIBUNWIND_MAX_PROCNAME_LENGTH characters.
+            // So we have a truncated procedure name that can not be demangled.
+            // We ignore the problem and the truncated non-demangled name will
+            // be printed.
+        } else {
+            vProcedureName[0] = '?';
+            vProcedureName[1] = '?';
+            vProcedureName[2] = '?';
+            vProcedureName[3] = 0;
+        }
+
+        // Resolve the object file name using dladdr:
+        if (dladdr((void*)(pip.start_ip + off), &dlinfo) && dlinfo.dli_fname && *dlinfo.dli_fname) {
+            vDynObjectFileName = dlinfo.dli_fname;
+        } else {
+            vDynObjectFileName = "???";
+        }
+
+        // Resolve the source file name using DWARF:
+        if (dwfl != NULL) {
+            addr = (uintptr_t)(ip - 4);
+            [[maybe_unused]] Dwfl_Module* module = dwfl_addrmodule(dwfl, addr);
+            // Here we could also ask for the procedure name:
+            //const char* vProcedureName = dwfl_module_addrname(module, addr);
+            // Here we could also ask for the object file name:
+            //vDynObjectFileName = dwfl_module_info(module, NULL, NULL, NULL, NULL, NULL, NULL, NULL);
+            vDWARFObjLine = dwfl_getsrc(dwfl, addr);
+            if (vDWARFObjLine != NULL) {
+                vSourceFileName = dwfl_lineinfo(vDWARFObjLine, &addr, &vSourceFileLineNumber, NULL, NULL, NULL);
+                //fprintf(stderr, " %s:%d", strrchr(vSourceFileName, '/')+1, vSourceFileLineNumber);
+            }
+        }
+        if (dwfl == NULL || vDWARFObjLine == NULL || vSourceFileName == NULL) {
+            vSourceFileName = "???";
+            vSourceFileLineNumber = 0;
+        }
+
+        // Print the stack frame number:
+        fprintf(stderr, "#%2d:", ++n);
+
+        // Print the stack addresses:
+        fprintf(stderr, " 0x%016" PRIxPTR " sp=0x%016" PRIxPTR, static_cast<uintptr_t>(ip), static_cast<uintptr_t>(sp));
+
+        // Print the source file name:
+        fprintf(stderr, " %s:%d", vSourceFileName, vSourceFileLineNumber);
+
+        // Print the dynamic object file name (that is the library name).
+        // This is typically not interesting if we have the source file name.
+        //fprintf(stderr, " %s", vDynObjectFileName);
```

Comment:
Why  comment this?

Reply:
this code refers to https://stackoverflow.com/a/45057441/3315536